From 1d60f339ecf786b8afe657e3aeb09baf1a35d253 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 6 Sep 2022 07:43:14 +0800
Subject: [PATCH 10/60] Revert "drm/exynos/exynos7_drm_decon: free resources
 when clk_set_parent() failed."

This reverts commit 5713b0be6dd94210413404dd8a71a40bb83654b1.
---
 drivers/gpu/drm/exynos/exynos7_drm_decon.c | 17 ++++-------------
 1 file changed, 4 insertions(+), 13 deletions(-)

diff --git a/drivers/gpu/drm/exynos/exynos7_drm_decon.c b/drivers/gpu/drm/exynos/exynos7_drm_decon.c
index 1c04c232dce15..f2d87a7445c73 100644
--- a/drivers/gpu/drm/exynos/exynos7_drm_decon.c
+++ b/drivers/gpu/drm/exynos/exynos7_drm_decon.c
@@ -800,40 +800,31 @@ static int exynos7_decon_resume(struct device *dev)
 	if (ret < 0) {
 		DRM_DEV_ERROR(dev, "Failed to prepare_enable the pclk [%d]\n",
 			      ret);
-		goto err_pclk_enable;
+		return ret;
 	}
 
 	ret = clk_prepare_enable(ctx->aclk);
 	if (ret < 0) {
 		DRM_DEV_ERROR(dev, "Failed to prepare_enable the aclk [%d]\n",
 			      ret);
-		goto err_aclk_enable;
+		return ret;
 	}
 
 	ret = clk_prepare_enable(ctx->eclk);
 	if  (ret < 0) {
 		DRM_DEV_ERROR(dev, "Failed to prepare_enable the eclk [%d]\n",
 			      ret);
-		goto err_eclk_enable;
+		return ret;
 	}
 
 	ret = clk_prepare_enable(ctx->vclk);
 	if  (ret < 0) {
 		DRM_DEV_ERROR(dev, "Failed to prepare_enable the vclk [%d]\n",
 			      ret);
-		goto err_vclk_enable;
+		return ret;
 	}
 
 	return 0;
-
-err_vclk_enable:
-	clk_disable_unprepare(ctx->eclk);
-err_eclk_enable:
-	clk_disable_unprepare(ctx->aclk);
-err_aclk_enable:
-	clk_disable_unprepare(ctx->pclk);
-err_pclk_enable:
-	return ret;
 }
 #endif
 
-- 
2.25.1

